testcases/kernel/controllers/pids/pids.sh

```diff
@@ -196,6 +209,118 @@ case5()
 	fi
 }
 
+case6()
+{
+	tst_res TINFO "set a limit that is smaller than current number of pids"
+	start_pids_tasks2 $max
+
+	lim=$((max - 1))
+	ROD echo $lim \> $testpath/pids.max
+
+	pids_task1 "$testpath/tasks"
+	ret=$?
+
+	if [ "$ret" -eq "2" ]; then
+		tst_res TPASS "fork failed as expected"
+	elif [ "$ret" -eq "0" ]; then
+		tst_res TFAIL "fork didn't fail despite the limit"
+	else
+		tst_res TBROK "pids_task1 failed"
+	fi
+
+	stop_pids_tasks
+}
+
+case7()
+{
+	tst_res TINFO "the number of all child cgroup tasks larger than its parent limit"
+	
+	num=10
+	lim=$((max / num))
+	for i in `seq 1 $num`; do
+		mkdir $testpath/child$i
+		ROD echo $lim \> $testpath/child$i/pids.max
+		start_pids_tasks2_path $testpath/child$i $lim
+	done
+
+	ROD echo $max \> $testpath/pids.max
+
+	pids_task1 "$testpath/tasks"
+	ret=$?
+
+	if [ "$ret" -eq "2" ]; then
+		tst_res TPASS "fork failed as expected"
+	elif [ "$ret" -eq "0" ]; then
+		tst_res TFAIL "fork didn't fail despite the limit"
+	else
+		tst_res TBROK "pids_task1 failed"
+	fi
+
+	for i in `seq 1 $num`; do
+		stop_pids_tasks_path $testpath/child$i
+		rmdir $testpath/child$i
+	done
+
+	stop_pids_tasks
+}
+
+case8()
+{
+	tst_res TINFO "set child cgroup limit smaller than its parent limit"
+	ROD echo $max \> $testpath/pids.max
+
+	mkdir $testpath/child
+
+	lim=$((max - 1))
+	ROD echo $lim \> $testpath/child/pids.max
+	tmp=$((max - 2))
+	start_pids_tasks2_path $testpath/child $tmp
+
+	pids_task1 "$testpath/child/tasks"
+	ret=$?
+
+	if [ "$ret" -eq "2" ]; then
+		tst_res TPASS "fork failed as expected"
+	elif [ "$ret" -eq "0" ]; then
+		tst_res TFAIL "fork didn't fail despite the limit"
+	else
+		tst_res TBROK "pids_task1 failed"
+	fi
+
+	stop_pids_tasks_path $testpath/child
+	rmdir $testpath/child
+	stop_pids_tasks_path
+}
```

Comment:
This test does not really make sense to me, we are just testing that the limit in child kicks in the parent is not involved at all.

Reply:
This test means the child cgroup is limited by its parent and own.

Reply:
I still fail to understand why do we even bother to set the parent limit, it's not like it's used. What happens here is that the smaller limits the child cgroup so the limit in the parent never limits anything. Or do I miss something?

What is exactly the point of this testcase?